packages/workflow/src/agent/agent-workflow.ts

```diff
@@ -57,6 +56,24 @@ export class AgentStepEvent extends WorkflowEvent<{
   toolCalls: AgentToolCall[];
 }> {}
 
+export type AgentWorkflowParams = {
+  /**
+   * List of agents to include in the workflow.
+   * Need at least one agent.
+   */
+  agents: BaseWorkflowAgent[];
+  /**
+   * The agent to start the workflow with.
+   * Must be an agent in the `agents` list.
+   */
+  rootAgent: string | BaseWorkflowAgent;
```

Comment:
if you allow `rootAgent` to be a `string` then `agents` should also be `BaseWorkflowAgent[] | string[]` for consistency.
What's the advantage of allowing string for rootAgent?

```suggestion
  agents: BaseWorkflowAgent[];
  /**
   * The agent to start the workflow with.
   * Must be an agent in the `agents` list.
   */
  rootAgent: BaseWorkflowAgent;
```

Reply:
if you keep string, you need to do for consistency:

```typescript 
  agents: string[] | BaseWorkflowAgent[];
  /**
   * The agent to start the workflow with.
   * Must be an agent in the `agents` list.
   */
  rootAgent: string | BaseWorkflowAgent;

Reply:
Hmm, the `agents` must be `BaseWorkflowAgent[]`